File: ticket-backend/src/main/java/aics/infrastructure/auth/LoggedUserDetailsDto.java
package aics.infrastructure.auth;

import aics.domain.user.RoleEnum;
import lombok.Data;
import lombok.experimental.Accessors;

import java.util.Set;

@Data
@Accessors(chain = true)
public class LoggedUserDetailsDto {
    private Long userId;
    private String name;
    private String sub;
    private String email;
    private String preferredUsername;
    private String givenName;
    private String familyName;
    private String fullName;
    private Set<RoleEnum> roles;

    public static LoggedUserDetailsDto fromLoggedUserDetails(LoggedUserDetails loggedUserDetails) {
        return new LoggedUserDetailsDto()
            .setUserId(loggedUserDetails.getUserId())
            .setName(loggedUserDetails.getName())
            .setSub(loggedUserDetails.getSub())
            .setEmail(loggedUserDetails.getEmail())
            .setPreferredUsername(loggedUserDetails.getPreferredUsername())
            .setGivenName(loggedUserDetails.getGivenName())
            .setFamilyName(loggedUserDetails.getFamilyName())
            .setFullName(loggedUserDetails.getFullName())
            .setRoles(loggedUserDetails.getRoles());
    }
}

File: ticket-frontend/src/modules/auth/logged-user-details-dto.ts
import { RoleEnum } from './role-enum';

export class LoggedUserDetailsDto {
    userId: number;
    name: string;
    sub: string;
    email: string;
    preferredUsername: string;
    givenName: string;
    familyName: string;
    fullName: string;
    roles: RoleEnum[];

    constructor(obj: {
        userId?: number,
        name?: string,
        sub?: string,
        email?: string,
        preferredUsername?: string,
        givenName?: string,
        familyName?: string,
        fullName?: string,
        roles?: RoleEnum[],
    }) {
        this.userId = obj.userId ? obj.userId : 0;
        this.name = obj.name ? obj.name : '';
        this.sub = obj.sub ? obj.sub : '';
        this.email = obj.email ? obj.email : '';
        this.preferredUsername = obj.preferredUsername ? obj.preferredUsername : '';
        this.givenName = obj.givenName ? obj.givenName : '';
        this.familyName = obj.familyName ? obj.familyName : '';
        this.fullName = obj.fullName ? obj.fullName : '';
        this.roles = obj.roles ? obj.roles : [];
    }
}